### DISCUSSION TITLE
x-bind:selected not working in Safari

### DISCUSSION THREAD

**Discussion by stefankempf** [alpinejs/alpine#3838] [1. Help] [answered]
I'm updating the selected option of a dropdown, based on current url parameters. It is working fine in all browsers except in safari. I'm doing this: 
```
function getUrlParameter(name) {
  const params = new URLSearchParams(window.location.search);
  return params.get(name);
}
```

and the markup looks like this: 
```
<div x-data='{ category: getUrlParameter("category") || "all"'>
  <select name="category" x-on:change="category = $el.value" >
    <option x-bind:selected="category === 'exampleCategory'">categoryTitle</option>
  </select>
</div>
```

Yet in Safari, the selection of the dropdown is not updating. Is that some kind of Bug, related to Safari?

**Comment by ekwoka** [comment]
The x-data expression isn't valid.

And use x-model on the select element. 👍

**Reply by stefankempf** [reply]
Thanks for your inputs. I've changed to x-model and that works fine in Safari as well.

```
<div x-data='{ category: getUrlParameter("category") || "all"'>
  <select name="category" x-on:change="category = $el.value" x-model="category">
    <option>categoryTitle</option>
  </select>
</div>
```

One thing I still don't get. Why is the x-data expression invalid? I get no error message and the function is working. I also update cards further down the dom with x-show and that feature works great as well.

**Reply by ekwoka** [reply]
The missing closing `}`

It will be a syntax error when the function is created.

You can remove the on change.

X-model is 2 way binding.

**Reply by stefankempf** [reply]
Ah yeah sure, sorry for that. I was striping out some complexity for ease of reading. forgot to close the x-data. Thanks again.

**Comment by stefankempf** [comment, marked as answer]
```
<div x-data='{ category: getUrlParameter("category") || "all" }'>
  <select name="category" x-model="category">
    <option>categoryTitle</option>
  </select>
</div>
```
This worked out great, many thanks @ekwoka
